ext/scheduler/airflow/airflow.go

```diff
@@ -146,6 +150,23 @@ func (s *Scheduler) DeployJobs(ctx context.Context, tenant tenant.Tenant, jobs [
 	return multiError.ToErr()
 }
 
+// uploadCommonLib is a function to upload the shared lib file to the bucket, encapsulated in a sync process.
+// we need to ensure that the file is uploaded only once, not on every job upload.
+func (s *Scheduler) uploadCommonLib(ctx context.Context, tenant tenant.Tenant, bucket Bucket) error {
+	if _, isSynced := s.hasLibSyncedMap.LoadOrStore(tenant.ProjectName().String(), true); isSynced {
+		return nil
+	}
+
+	err := bucket.WriteAll(ctx, filepath.Join(jobsDir, baseLibFileName), SharedLib, nil)
+	if err != nil {
+		// on failure, flag is removed so that the upload is retried in the next job upload
+		s.hasLibSyncedMap.Delete(tenant.ProjectName().String())
```

Comment:
Current flow is not intuitive, we are loading or storing above and delete here. Can we simplify this ?
Use only Load above to check the existence, and here we just store if it was successful.

Reply:
My consideration is with this behavior, we can ensure that if there are multiple parallel calls to `DeployJobs` with the same project, only one request gets to call `bucket.WriteAll`.
The rest will return on the LoadAndStore check because the flag is already set by the successful call.

Reply:
Sure, thank you.